src/minerva_lib/blend.py

```diff
@@ -1,102 +1,70 @@
-        raise ValueError('Scaling to 0,1 requires unsigned integers')
-
-    one = dtype_info.max
-    return np.float32(img / one)
-
-
-def f32_to_bgr(f_img, color=[1, 1, 1]):
-    '''Reshape into a color image
-
     Arguments:
-    f_img: float32 image to reshape
+        channel: Dict to blend with rendering settings:
+            {
+                image: Numpy image data
+                min: Range minimum, float32 range 0, 1
+                max: Range maximum, float32 range 0, 1
+            }
+
+    Returns:
+        Threshholded float32 image normalzied within 0, 1
+        r, g, b float32 array color within 0, 1
     '''
+    image = channel['image']
+    min_ = channel['min']
+    max_ = channel['max']
 
-    # All inputs should be normalized between 0 and 1
-    try:
-        assert np.all((f_img >= 0) & (f_img <= 1))
-    except AssertionError:
-        raise ValueError('Color image requires values from 0,1')
+    # Return image for additive blending
+    f32_image = skimage.img_as_float(image)
+    f32_image = skimage.exposure.rescale_intensity(f32_image, (min_, max_))
 
-    # Give the image a color dimension
-    f_vol = f_img[:, :, np.newaxis]
-    f_bgr = np.repeat(f_vol, 3, 2) * color
-    return np.round(255 * f_bgr).astype(np.uint8)
+    return f32_image, channel['color']
 
 
-def linear_bgr(channels):
+def linear_rgb(channels):
     '''Blend all channels given
     Arguments:
         channels: List of dicts of channels to blend with rendering settings:
             {
                 image: Numpy image data
-                color: N-channel by b, g, r float32 color
-                min: Range minimum, float32 range
-                max: Range maximum, float32 range
+                color: r, g, b float32 array within 0, 1
+                min: Range minimum, float32 range 0, 1
+                max: Range maximum, float32 range 0, 1
             }
 
     Returns:
-        uint8 y by x by 3 color BGR image
+        float32 y by x by r, g, b color image within 0, 1
     '''
+    num_channels = 0
 
-    # Get the number of channels
-    num_channels = len(channels)
+    # threshhold images and normalize colors
+    for image, color in map(handle_channel, channels):
 
-    # Must be at least one channel
-    if num_channels < 1:
-        raise ValueError('At least one channel must be specified')
-
-    # Ensure that dimensions of all channels are equal
-    shape = channels[0]['image'].shape
-    for channel in channels:
-        if channel['image'].shape != shape:
-            raise ValueError('All channel images must have equal dimensions')
-
-    # Shape of 3 color image
-    shape_color = shape + (3,)
-
-    # Final buffer for blending
-    image_buffer = np.zeros(shape_color, dtype=np.float32)
+        num_channels += 1
+        if num_channels is 1:
+            # Compatible with images, vectors, or single pixels
+            in_shape = iter(image.shape)
+            out_shape = [next(in_shape, i) for i in [1, 1, 3]]
```

Comment:
Here it's more complicated than `out_shape = image.shape + (3,)` because `handle_channel` will remove singleton dimensions from input images like this `[[0]]` or this `[[0, 0]]`.